### PR TITLE
Use VoteProgram account as staking account

### LINKED ISSUE #2551: Identifying staking accounts is overly complex
#### Problem

We currently identify validator staking accounts by querying all accounts for ones with program id `VoteProgram::id()`. Then we deserialize those accounts and grab its field `node_id`. That `node_id` is a pubkey. Its account balance is assumed to the validator's stake.

That extra level of indirection appears to serve no purpose. Furthermore, the `VoteSigner` only guards the private key of `VotePrograms`, not its `node_id`.

#### Proposed Solution

Use the `VoteProgram` account as the staking account. Here's the first attempt:

https://github.com/solana-labs/solana/pull/2539

### PR DESCRIPTION
**Pull request by garious**
#### Problem

When we create the VoteProgram account, we point it to the pubkey of a staking account, but why the extra level of indirection? It's the VoteProgram account that's guarded by the VoteSigner, not the validator's account.

#### Summary of Changes

* Remove node_id
* Update tests to stop creating an extra account just to point the VoteProgram to

TODO:
- [ ] Fix the other dozen tests this breaks

Fixes #2551

### CODE REVIEW

**Comment by garious** [comment]
@carllin, @pgarg66, fyi

@aeyakovenko, is it okay for stakers to trust VoteProgram?

**Comment by aeyakovenko** [comment]
@garious What is the level of trust for Stakers?  The main things for main net
* validators self generate keys on boot. These keys are ephemeral and are expected to be rotated quickly.
* Stake holders with a human operator assign their stake to a validator.  Stake holder keys are not ephemeral and have custodial storage.
* The rewards generated by a validator can be claimed by a human stake holder, and not the validators self generated key.

**Comment by aeyakovenko** [comment]
@garious you can move forward however you feel would speed things along, but the main thing is that for main net, the human custody of the stake needs to be separate from the operation keys that validators use.  The human custody may require M/N setup as well.

File: src/leader_scheduler.rs [hunk 1]
```diff
@@ -709,23 +708,16 @@ pub mod tests {
...
-            let pk = new_keypair.pubkey();
-            old_ids.insert(pk.clone());
-
-            // Give the account some stake
-            bank.transfer(5, &mint_keypair, pk, genesis_block.last_id())
-                .unwrap();
-
             // Create a vote account
-            let vote_signer = VoteSignerProxy::new_local(&new_keypair);
+            let vote_signer = VoteSignerProxy::new();
```
Comment (pgarg66) [inline src/leader_scheduler.rs]: Is there a VoteSignerProxy::new()? I thought you last PR changed it to new_with_signer()

Comment (garious) [reply src/leader_scheduler.rs]: Yes, I implemented the KeypairUtil trait, which awkwardly defines a `new()`. I pointed it to LocalVoteSigner. By using it here, we're saying that the client-side keypair is ephemeral and not a staking account.

Comment (garious) [reply src/leader_scheduler.rs]: The reason I renamed `new(...)` to `new_with_signer(...)` was to make space fo the `KeypairUtil` version, which takes no arguments.

Comment (pgarg66) [reply src/leader_scheduler.rs]: 👍 

It looks good from vote signing point of view. Also, I think the staking account is correct (same as voting account). @carllin should also review this part.

**Comment by garious** [comment]
This PR caused too much trouble, abandoning. The main issue turned out to be the change of using the VoteProgram's pubkey for leader rotation instead of `VoteProgram::node_id`. Current code makes extensive use of that `node_id` and its private key.  The most notable is the signing of blocks when the node is the leader. If we continued in the direction set by this PR, the `VoteSignerProxy` would need to sign blocks.  Instead, it's quite reasonable for a staker to delegate that responsibility to some less secure Fullnode keypair, as is currently implemented.

Some of the functionality in this PR will return again soon. Stake should be held by the `VoteProgram`, not by `VoteProgram::node_id`. All the token transfers to `VoteProgram::node_id` are unnecessary. Those accounts should not be trusted with staked tokens.

**Comment by aeyakovenko** [comment]
@garious, There are three main problems. 

1. Securing the fullnode key from the untrusted network connected code.

2. Delegating stakes to a key that signs blocks.  This separates the ownership of stakes from the keys that are live and voting with those stakes. 

3. Preventing a voting key from getting slashed even if it is receiving compromised requests. 


My understanding was that the proxy was solving 1 and 3, and the staking program was solving 2.

**Pull request closed without merging**